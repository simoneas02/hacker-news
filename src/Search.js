import React from 'react';

const Search = ({ value, onChange, children }) =>
    <form>
        { children }
        <input 
            type='text'
            onChange={ onChange }
            value={ value }
        />
    </form>

export default Search;