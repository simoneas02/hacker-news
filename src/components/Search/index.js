import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Search extends Component {
  componentDidMount() {
    if (this.input) {
      this.input.focus();
    }
  }

  render() {
    const { value, onChange, onSubmit, children } = this.props;
    return (
      <form onSubmit={onSubmit}>
        {children}
        <input
          type="text"
          onChange={onChange}
          value={value}
          ref={node => (this.input = node)}
        />
        <button type="submit">{children}</button>
      </form>
    );
  }
}

Search.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
  onSubmit: PropTypes.func,
  children: PropTypes.node
};

export default Search;
