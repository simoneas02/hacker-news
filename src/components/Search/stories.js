import React from 'react';
import { storiesOf } from '@storybook/react';

import Search from '.';

storiesOf('Search', module).add('Basic search', () => (
  <Search value="redux" children="Search" />
));
