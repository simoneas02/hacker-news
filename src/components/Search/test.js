import React from 'react';
import ReactDOM from 'react-dom';
import Search from '.';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Search />, div);
  ReactDOM.unmountComponentAtNode(div);
});
