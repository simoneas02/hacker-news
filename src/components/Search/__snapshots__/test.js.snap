// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<form>
  <input
    type="text"
  />
  <button
    type="submit"
  />
</form>
`;
